doc/command_guidelines.md

```diff
@@ -0,0 +1,63 @@
+# Command Guidelines
+
+This document describes the command guidelines for 'az' and applies to both CLI command modules and extensions.
+
+Guidelines marked (*) only apply to command modules, not extensions.
+
+If in doubt, ask!
+
+## General Patterns
```

Comment:
It seems like some of the stuff in General Patterns would better be placed under other subheadings. I call out a few but best to consider them all.

Reply:
I think a separation between 'general patterns' and other things that pertain only to 'command naming' etc. makes sense.
Arguably everything could be put under 'Behavior' but keeping each section a few points each makes things clearer I think.